LiteDB/Engine/Disk/DiskWriterQueue.cs

```diff
@@ -100,14 +102,23 @@ private async Task ExecuteQueue()
 
                         _stream.FlushToDisk();
 
-                        await _queueHasItems.WaitAsync();
+                        await _queueHasItems.WaitHandle.WaitAsync().ConfigureAwait(false);
                     }
                 }
             }
             catch (Exception ex)
             {
-                _state.Handle(ex);
+                _state.Handle(LiteException.InvalidDatafileState(ex, "DiskWriterQueue failed"));
                 _exception = ex;
+                ExhaustQueue();
+            }
+        }
+
+        private void ExhaustQueue()
+        {
+            while (_queue.TryDequeue(out var page))
+            {
+                page.Release();
             }
         }
```

Comment:
If anything fails here we need to drop this object - this means all pages must be released (including the one that caused the failure - see the finally case below)